File: Finding Array of Longest Prefix Sum/code.py
def computeLPS(s):
    n = len(s)
    j = 0
    lps = [0] * n
    for i in range(1, n):
        while j > 0 and s[i] != s[j]:
            j = lps[j - 1]
        if s[i] == s[j]:
            j += 1
        lps[i] = j
    return lps

def main():
    s = "abacab"
    lps = computeLPS(s)
    print("LPS Array:", " ".join(map(str, lps)))

if __name__ == "__main__":
    main()

File: Finding Array of Longest Prefix Sum/code.java
import java.util.*;

public class LPSArray
{
    public static int[] computeLPS(String s)
  {
        int n = s.length();
        int j = 0;
        int[] lps = new int[n];
        for(int i = 1; i < n; i++)
        {
            while(j > 0 && s.charAt(i) != s.charAt(j))
            {
                j = lps[j - 1];
            }
            if(s.charAt(i) == s.charAt(j))
            {
                j++;
            }
            lps[i] = j;
        }
        return lps;
    }

    public static void main(String[] args) {
        String s = "abacab";
        int[] lps = computeLPS(s);
        System.out.print("LPS Array: ");
        for (int x : lps) {
            System.out.print(x + " ");
        }
        System.out.println();
    }
}
